### PR TITLE
TC-SC-4.10 python test

### PR DESCRIPTION
**Pull request by raul-marquez-csa**
**Category:**
Functional conformance

**Description:**
The purpose of this test case is to verify that a Short Idle Time ICD node properly advertises SAI/SII
values.

**Verifies:**
  - SII key is higher than the SESSION_IDLE_INTERVAL default value (500 milliseconds)
  - SII key and SAI key is less than 1 hour (3600000 milliseconds)

**Full test plan link for details:**
https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/securechannel.adoc#tc-sc-4-10-operational-discovery-sit-icd-node-dut_commissionee

### CODE REVIEW

File: src/python_testing/mdns_helper.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        pass
+
+    def remove_service(self, zc: Zeroconf, type: str, name: str) -> None:
+        pass
+
+    def update_service(self, zc: Zeroconf, type: str, name: str) -> None:
+        pass
+
+
+class MdnsHelper:
```
Comment (cecille) [inline src/python_testing/mdns_helper.py:41]: Can you change the name of this to indicate this is for operational?

I think some of this logic will end up being common between operational and commissionable, so I think eventually you'll end up with a generic base class that exposes an operational and commissionable (and maybe commissioner) option. Anyway, for now, if you just change the name, you can extract the common bits later.

File: src/python_testing/mdns_helper.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    def __init__(self, dut):
```
Comment (cecille) [inline src/python_testing/mdns_helper.py:42]: what is dut here? Just the ID? Can you add a type indicator maybe a doc comment?

File: src/python_testing/mdns_helper.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    def _get_txt_record_key_value(self, key: str):
+        # Convert the key from string to bytes, as the dictionary uses bytes
+        byte_key = key.encode('utf-8')
+
+        # Check if the key exists in the dictionary
+        asserts.assert_in(byte_key, self._service_info.properties, f"Property '{key}' not found")
+        value = self._service_info.properties[byte_key]
+
+        # Check if the value is not None
+        asserts.assert_is_not_none(value, f"Value for property '{key}' not present")
```
Comment (cecille) [inline src/python_testing/mdns_helper.py:67]: Can you please move this external to the class? That will make this class more generally useful for other tests in the future.

File: src/python_testing/mdns_helper.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    async def getTxtRecord(self, key: str = None, refresh=False):
+        """
+        Asynchronously retrieves the TXT record for a given device.
+
+        This method queries the MDNS service to find the TXT record associated with
+        the specified device. It handles the lookup process and extracts specific
+        property values based on the provided key.
+
+        Args:
+            dut: The device under test. It should contain attributes like `dut_node_id` and
```
Comment (cecille) [inline src/python_testing/mdns_helper.py:81]: dut isn't passed in as a parameter here.

File: src/python_testing/mdns_helper.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        asserts.assert_in(MDNS_TYPE_OPERATIONAL, self._service_types,
+                          f"The MDNS operational service type '{MDNS_TYPE_OPERATIONAL}' was not found")
+
+        # Fetch service info and properties (TXT) and cache them
+        if self._service_info is None or refresh:
+            service_info: AsyncServiceInfo = await self._azc.async_get_service_info(
+                name=self._name,
+                type_=MDNS_TYPE_OPERATIONAL
+            )
+            asserts.assert_is_not_none(service_info, f"Service info for {self._name} not found")
```
Comment (cecille) [inline src/python_testing/mdns_helper.py:156]: same here.

File: src/python_testing/mdns_helper.py
Comment (andy31415) [inline src/python_testing/mdns_helper.py:1]: Please avoid names that contain filler words such as helper/support/util or similar.

Could we define what this file intends to do without such naming? This applies to both file name and class names. maybe something like `MatterMdnsRecordValidation` for class name (and translate this into a filename as well) or just mdns_validation or similar.

File: src/python_testing/mdns_helper.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+#
+
+from mobly import asserts
+from zeroconf import ServiceListener, Zeroconf
+from zeroconf.asyncio import AsyncServiceInfo, AsyncZeroconf, AsyncZeroconfServiceTypes
+
+MDNS_TYPE_OPERATIONAL = "_matter._tcp.local."
+
+'''
+A service listener for the TXT record data to get populated
```
Comment (andy31415) [inline src/python_testing/mdns_helper.py:25]: doc comments should follow class definition. How about `EmptyServiceListener` or `NoopServiceListener` ?

File: src/python_testing/mdns_helper.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        asserts.assert_in(byte_key, self._service_info.properties, f"Property '{key}' not found")
+        value = self._service_info.properties[byte_key]
+
+        # Check if the value is not None
+        asserts.assert_is_not_none(value, f"Value for property '{key}' not present")
+
+        # Convert the value from bytes to string
+        return value.decode('utf-8')
+
+    async def getTxtRecord(self, key: str = None, refresh=False):
```
Comment (andy31415) [inline src/python_testing/mdns_helper.py:72]: could we type the return value as well?

Comment (andy31415) [reply src/python_testing/mdns_helper.py:72]: Type of key should be `Optional[str]` if the value can be none

File: src/python_testing/TC_SC_4_10.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+        # *** Step 1 ***
+        self.print_step(1, "DUT is instructed to advertise its service: already done")
+
+        # *** Step 2 ***
+        self.print_step(2, "TH scans for DNS-SD advertising, looks for SAI/SII values")
+
+        # Get SAI/SII values
+        mh = MdnsHelper(self)
+        SAI_MS = int(await mh.getTxtRecord(key="SAI"))
```
Comment (andy31415) [inline src/python_testing/TC_SC_4_10.py:57]: I would lowercase these as `sai_ms` and `sii_ms` since these are not constants (as opposed to the others)

**Review by raju-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by raul-marquez-csa** [comment]
This test case has been deprecated and has been included in https://github.com/project-chip/connectedhomeip/pull/31982

**Pull request closed without merging**